### PR TITLE
test: reducing H2 test permutations, increasing coverage time

### PR DESCRIPTION
**Pull request by alyssawilk**
The settings for edge include

#define HTTP2SETTINGS_EDGE_COMBINE                                                                 \ 
  ::testing::Combine(                                                                              \ 
      ::testing::Values(Http2Settings::MIN_HPACK_TABLE_SIZE, Http2Settings::MAX_HPACK_TABLE_SIZE), \ 
      ::testing::Values(Http2Settings::MIN_MAX_CONCURRENT_STREAMS,                                 \ 
                        Http2Settings::MAX_MAX_CONCURRENT_STREAMS),                                \ 
      ::testing::Values(Http2Settings::MIN_INITIAL_STREAM_WINDOW_SIZE,                             \ 
                        Http2Settings::MAX_INITIAL_STREAM_WINDOW_SIZE),                            \ 
      ::testing::Values(Http2Settings::MIN_INITIAL_CONNECTION_WINDOW_SIZE,                         \ 
                        Http2Settings::MAX_INITIAL_CONNECTION_WINDOW_SIZE))

The existing codec unit tests don't appear to cover large bodies or many streams, so I think all we need is one test to make sure these settings work.  I've done this with the codec header compression test as it seems relevant to hpack table size.

This reduces test run time 8x, and should claw back about 14% of our coverage run time.

Sadly it does not reduce it enough so also bumping coverage time.

Risk Level: n/a
Testing: unit test still passes
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/common/http/http2/codec_impl_test.cc [hunk 1]
```diff
@@ -353,7 +353,7 @@ TEST_P(Http2CodecImplTest, TrailingHeaders) {
   response_encoder_->encodeTrailers(TestHeaderMapImpl{{"trailing", "header"}});
 }
 
-TEST_P(Http2CodecImplTest, TrailingHeadersLargeBody) {
+TEST_P( Http2CodecImplTest, TrailingHeadersLargeBody) {
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply test/common/http/http2/codec_impl_test.cc]: outdated :-)

**Comment by alyssawilk** [comment]
agree a comment would be good.  would you either be Ok with a follow up or force merging so I don't have to wait 2 more hours for coverage to finish?

**Comment by htuch** [comment]
Force merge is good if we have a successful run now.

**Comment by alyssawilk** [comment]
Nope :-(
Adding comment and bumping time because I'm sick of these failures.

**Pull request merged**